src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -371,16 +368,62 @@ private async ValueTask<EvaluationEvent> EvaluateFeature<TContext>(string featur
                     await sessionManager.SetAsync(evaluationEvent.FeatureDefinition.Name, evaluationEvent.Enabled).ConfigureAwait(false);
                 }
 
-                if (evaluationEvent.FeatureDefinition.Telemetry != null &&
-                    evaluationEvent.FeatureDefinition.Telemetry.Enabled)
+                // Only add an activity event if telemetry is enabled for the feature
+                if (telemetryEnabled)
                 {
-                    PublishTelemetry(evaluationEvent, cancellationToken);
+                    AddEvaluationActivityEvent(evaluationEvent);
                 }
             }
 
             return evaluationEvent;
         }
 
+        private void AddEvaluationActivityEvent(EvaluationEvent evaluationEvent)
+        {
+            // Break if there is no reason to emit an event
+            if (Activity.Current == null || !Activity.Current.IsAllDataRequested)
+            {
+                return;
+            }
+
+            ActivityTagsCollection tags = new ActivityTagsCollection
+            {
+                { "FeatureName", evaluationEvent.FeatureDefinition.Name },
+                { "Enabled", evaluationEvent.Enabled },
+                { "VariantAssignmentReason", evaluationEvent.VariantAssignmentReason },
+                { "Version", ActivitySource.Version }
+            };
+
+            if (evaluationEvent.TargetingContext is { UserId: string userId })
+            {
+                tags["TargetingId"] = userId;
+            }
+
+            if (evaluationEvent.Variant is { Name: string variantName })
+            {
+                tags["Variant"] = variantName;
+            }
+
+            if (evaluationEvent.FeatureDefinition.Telemetry.Metadata is { } metadata)
+            {
+                foreach (KeyValuePair<string, string> kvp in metadata)
+                {
+                    if (tags.ContainsKey(kvp.Key))
+                    {
+                        Logger?.LogWarning($"{kvp.Key} from TelemetryMetadata will be ignored, as it would override an existing key.");
```

Comment:
```suggestion
                        Logger?.LogWarning($"{kvp.Key} from telemetry metadata will be ignored, as it would override an existing key.");
```